### ISSUE TITLE
[Bug]: Inference on npu occure oom

### ISSUE THREAD

**Issue by pockers21** [openvinotoolkit/openvino#25626]
### OpenVINO Version

2024.2.0

### Operating System

Other (Please specify in description)

### Device used for inference

NPU

### Framework

PyTorch

### Model used

qwen2

### Issue description

OS:  Debian
Mem: 250G
hello_query_device result:
```
CPU:
DEVICE_ARCHITECTURE : intel64
FULL_DEVICE_NAME : Intel(R) Core(TM) Ultra 9 185H

GPU:
FULL_DEVICE_NAME : Intel(R) Graphics [0x7d55] (iGPU)
DEVICE_ARCHITECTURE : GPU: vendor=0x8086 arch=v12.70.0

NPU:
FULL_DEVICE_NAME : Intel(R) AI Boost
DEVICE_ARCHITECTURE : 3720
```

We fitune qwen2 model 7b with our dataset and use optimum-cli convert  to openvino ir format,  finally the size of qwen2  model in openvino was 7.2G.
Running code as below 

```
import torch
import openvino as ov

core = ov.Core()
core.add_extension("/home/user/miniforge3/envs/test/lib/python3.11/site-packages/openvino_tokenizers/lib/libopenvino_tokenizers.so")
base = "/home/user/openvino_test/openvino_model_7b/openvino/"
ov_model = core.read_model(base + "openvino_model.xml")
# Assign dynamic shapes to second dimension in every input layer
shapes = {}
for input_layer in ov_model.inputs:
    shape = input_layer.get_partial_shape()
    shape = [1,4]
    shapes[input_layer] = shape
ov_model.reshape(shapes)
import openvino.properties.hint as hints
config = {hints.performance_mode: hints.PerformanceMode.THROUGHPUT,
            hints.inference_precision: ov.Type.f32,
            streams.num: 1}

compile_model = core.compile_model(ov_model, "GPU")
print('compile model finish')
tokenizer =  core.compile_model(core.read_model(base + "openvino_tokenizer.xml"))
encoded_input = tokenizer(["hello"])


out = compile_model({"input_ids":encoded_input["input_ids"],"attention_mask":encoded_input["attention_mask"]})
print(f'out: {out}')
```
switch to every kind of device, CPU mode works well,  but GPU mode will print out as nan.

> out: {<ConstOutput: names[logits] shape[1,4,151936] type: f32>: array([[[nan, nan, nan, ..., nan, nan, nan],
>         [nan, nan, nan, ..., nan, nan, nan],
>         [nan, nan, nan, ..., nan, nan, nan],
>         ...,
>         [nan, nan, nan, ..., nan, nan, nan],
>         [nan, nan, nan, ..., nan, nan, nan],
>         [nan, nan, nan, ..., nan, nan, nan]]], dtype=float32)}

But while add config like this 
```
core.compile_model(ov_model, "GPU", config)
```
out value become normal.

l have noticed memory usage by top command, while config as THROUGHPUT, memory usage increase more slowly, what is the reason?

While using  NPU mode,  oom would be occured,  no matter config added or not.

Thks.

### Issue submission checklist

- [X] I'm reporting an issue. It's not a question.
- [X] I checked the problem with the documentation, FAQ, open issues, Stack Overflow, etc., and have not found a solution.
- [X] There is reproducer code and related data files such as images, videos, models, etc.
